sim/battle.js

```diff
@@ -1548,6 +1548,19 @@ class Battle extends Dex.ModdedDex {
 					} else {
 						pokemon.lastAttackedBy = null;
 					}
+					if (this.gen >= 7) pokemon.knownType = true; // If it was an illusion, it's not any more
+				}
+
+				if (this.gen >= 7) {
+					// In Gen 7, the real type of every Pokemon is visible to all players via the bottom screen while making choices
```

Comment:
Every turn? Ick. Why not only send it for weird type changes? Or just at the moment of the type change; in practice that's unlikely to be an info leak.